src/util/misc.js

```diff
@@ -176,7 +178,8 @@
      * @returns {Point} vector representing the unit vector of pointing to the direction of `v`
      */
     getHatVector: function (v) {
-      return new fabric.Point(v.x, v.y).scalarMultiply(1 / Math.hypot(v.x, v.y));
+      var hypot = Math.sqrt(v.x * v.x + v.y * v.y);
+      return new fabric.Point(v.x, v.y).scalarMultiply(1 / hypot);
```

Comment:
removed the remaining use of `Math.hypot`

Reply:
The commit message is wrong. I thought firefox calculated it wrong but it's chrome.
I wonder if there are other browser core methods that cause strange bugs in fabric.
I would never even think of checking `Math.hypot` or `Math.acos` assuming for sure they are correct.
Alarming notion.

Reply:
Coming to think of it, Tylor is probably how `hypot` is calculated. And then it is a question of the error precision.
But I'm guessing here